### ISSUE TITLE
enable persistent systemd journal

### ISSUE THREAD

**Issue by jankatins** [debops/debops#260]
For that to happen either the `/var/log/journal` directory has to exist (with the default debian systemd packages) or `Storage=persistent` has to go into a conf file (in `/etc/systemd/journal.conf.d/whatever.conf`). It would also be nice to have a setting for the size of the journal.

**Comment by drybjed** [comment]
You can create the directory using `debops.resources`:
```yaml
resources__paths:
  - name: '/var/log/journal'
    state: 'directory'
    owner: 'root'
    group: 'root'
    mode: '0750'
```
You could also create the configuration with the same role.

I'm not sure if restart/reload of `journald` is needed to take effect.

Since Debian Stretch uses `rsyslog` for persistent log files by default, I'm not sure if essentially duplicating the functionality with persistent journal by default is necessary. On the other hand, since you are using Debian Sid, do you know if there are plans to drop `rsyslog` in Buster?

**Comment by jankatins** [comment]
I actually want to disable rsyslogd. I use debops on my laptop and try to find all the knobs to configure it just like I have it now. journald instead of rsyslogd is something I had and I was kind of surprised that debops doesn't have a journal specifc role or at least some knobs in one of the roles. 

Re Debians plans: I don't think there are any plans to drop anything :-) rsyslogd is probably the right tool if you plan on forwarding logs and so on. But for a laptop (and even on a small server without central logging), I feel that journald is the better solution.

**Comment by drybjed** [comment]
You should be able to disable `rsyslog` **role** by setting
```yaml
rsyslog__enabled: False
```
in the inventory. However that will not uninstall `rsyslog` itself.

There's no `journald` role since for the time being I was busy maintaining the DebOps project itself, and I usually focus on the roles that I currently need at work. Since `rsyslog` is installed in Debian by default, I decided to focus on it and that's how `debops.rsyslog` happened. I haven't played with `journald` much - does it support sending logs to remote hosts yet, or is that still done via `rsyslog`, or some agent installed on the host? Ah, there's `systemd-journal-remote` package, maybe that's how its done.

If you want, you could write a `journald` Ansible role (there's no APT package with that name, which is usually the basis of naming DebOps roles, so it's free to use) and propose to add it to the project.

**Comment by jankatins** [comment]
Sending logs to remote is AFAIK using some rsyslog API. But I haven't played with that.

I just like stuff like `journalctl -u my-systemd-service` (logs of one service) and `journalctl -ef` (errors as they come in) :-)

I will try writing a simple role for that. What's the expectation for such a role, does such a role need to cover "everything" or is a "it works on my laptop" setting ok? Also, is it ok to use some of the already existing roles in the ansible Galaxy?

**Comment by drybjed** [comment]
You could start by reading [DebOps Guidelines and Policies](https://docs.debops.org/en/latest/debops-policy/index.html) section of the old documentation (still needs to be moved to a new layout). Since it's pretty big, you can focus on:
- [Software Source Policy](https://docs.debops.org/en/latest/debops-policy/software-source-policy.html) - where DebOps pulls code from
- [Code Standards Policy](https://docs.debops.org/en/latest/debops-policy/code-standards-policy.html) - how DebOps roles are written

A simple role for `journald` that covers the basics is fine. Proper documentation is of course, a must. Remember that the role is public and don't include any personal-focused configuration like your usernames, etc. You can use some of the Ansible local facts as sources of useful data, for example check out the [`debops.core` role guides](https://docs.debops.org/en/latest/ansible/roles/debops.core/guides.html).

DebOps has multiple local tests you can perform to check the documentation build, shell scripts, Python scripts, YAML files, etc. Run the `make help` command in the root of the monorepo to see available commands. You might need to install required software to do this manually.

DebOps uses the GPLv3 license, I hope that you are fine with that.

You can use roles from Ansible Galaxy as inspiration, but you will probably need to modify at least some code - for example, DebOps role variables are namespaced with `<role_name>__` prefix. Also, it's best to use existing dependent roles like `debops.ferm`, `debops.nginx`, etc. to manage other resources than the one your role manages, this keeps the project consistent.

You can publish the role in your fork of DebOps monorepo in a branch, you don't need to create a PR right away but that won't hurt either, and it can help review the role before it's merged. All PRs are also tested on Travis-CI, and from time to time I'm testing them via GitLab CI when the support is enabled.

**Comment by drybjed** [comment]
The [`journald` role](https://docs.debops.org/en/master/ansible/roles/journald/index.html#debops-journald) has been created and it will be included in the Debops v2.1.0 release; it's also available in the `master` branch. I plan to update the `rsyslog` role to support uninstallation of the `rsyslog` package, probably before DebOps v2.1.0 is released.
